src/ResourceManager/Compute/Commands.Compute/Extension/VmssDiskEncryption/GetAzureVmssDiskEncryptionStatus.cs

```diff
@@ -155,36 +155,98 @@ private PSVmssDiskEncryptionStatusContext GetVmssDiskStatus(string rgName, strin
             {
                 return psResult;
             }
-
             psResult.EncryptionExtensionInstalled = true;
 
+            // retrieve public configuration settings for the extension
             psResult.EncryptionSettings = JsonConvert.DeserializeObject<AzureVmssDiskEncryptionExtensionPublicSettings>(
                 ext.Settings.ToString());
 
-            if (psResult.EncryptionSettings.EncryptionOperation.Equals(AzureDiskEncryptionExtensionConstants.enableEncryptionOperation, StringComparison.OrdinalIgnoreCase))
-            {
-                psResult.EncryptionEnabled = true;
-            }
-
+            // retrieve any status summary for the extension 
             var vmssInstanceView = this.VirtualMachineScaleSetClient.GetInstanceView(rgName, vmssName);
-
             if (vmssInstanceView.Extensions == null
                 || vmssInstanceView.Extensions.Count == 0)
             {
                 return psResult;
             }
-
             try
             {
                 extSummary = vmssInstanceView.Extensions.First(e => e.Name.Equals(this.ExtensionName));
             }
             catch (InvalidOperationException)
             {
                 return psResult;
-            }
-
+            }            
             psResult.EncryptionSummary = extSummary.StatusesSummary;
 
+            // retrieve per disk information from all vm instances in the scale set
+            try
+            {
+                // retrieve the list of vm instances in the scale set 
```

Comment:
Does this change the output type?

Reply:
it does, but on closer review the change can be reverted, prepping a change to revert this

Reply:
@markcowl backed out the change to output type (same information can still be retrieved on a per instance basis if needed using the other cmdlet)